traffic_ops/traffic_ops_golang/profile/profile_import_test.go

```diff
@@ -0,0 +1,337 @@
+				param1,
+				param2,
+			},
+			returnedNewParameters:      1,
+			returnedExistingParameters: 1,
+		},
+		{
+			description: "Success: Dup of existing",
+			mockStorageReturns: map[string][]mockStorageReturn{
+				*param1.Name: []mockStorageReturn{
+					// Select Returns
+					mockStorageReturn{
+						empty: true,
+						query: "SELECT",
+					},
+					// Insert Returns
+					mockStorageReturn{
+						id:    1,
+						query: "INSERT INTO parameter",
+					},
+				},
+				*param2.Name: []mockStorageReturn{
+					// Select Returns
+					mockStorageReturn{
+						query: "SELECT",
+						id:    2,
+					},
+				},
+			},
+			parameters: []tc.ProfileExportImportParameterNullable{
+				param1,
+				param2,
+				param2,
+			},
+			returnedNewParameters:      1,
+			returnedExistingParameters: 1,
+		},
+		{
+			description: "Fail: Storage error selecting param",
+			mockStorageReturns: map[string][]mockStorageReturn{
+				*param1.Name: []mockStorageReturn{
+					// Select Returns
+					mockStorageReturn{
+						empty:      true,
+						query:      "SELECT",
+						storageErr: errors.New("storage error"),
+					},
+				},
+			},
+			parameters: []tc.ProfileExportImportParameterNullable{
+				param1,
+			},
+			returnedErr: true,
+		},
+		{
+			description: "Fail: Storage error inserting param",
+			mockStorageReturns: map[string][]mockStorageReturn{
+				*param1.Name: []mockStorageReturn{
+					// Select Returns
+					mockStorageReturn{
+						empty: true,
+						query: "SELECT",
+					},
+					// Insert Returns
+					mockStorageReturn{
+						query:      "INSERT INTO parameter",
+						storageErr: errors.New("storage error"),
+					},
+				},
+			},
+			parameters: []tc.ProfileExportImportParameterNullable{
+				param1,
+			},
+			returnedErr: true,
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			t.Log("Starting test scenario: ", testCase.description)
+			mockDB, mock, err := sqlmock.New()
+			if err != nil {
+				t.Fatalf("an error '%s' was not expected when opening a stub database connection", err)
+			}
+			defer mockDB.Close()
+			db := sqlx.NewDb(mockDB, "sqlmock")
+			defer db.Close()
+			mock.ExpectBegin()
+			for _, param := range testCase.parameters {
+				mockStorageReturns := testCase.mockStorageReturns[*param.Name]
+				for _, msr := range mockStorageReturns {
+					if msr.storageErr != nil {
+						mock.ExpectQuery(msr.query).WillReturnError(msr.storageErr)
+					} else {
+						rows := sqlmock.NewRows(idRow)
+						if !msr.empty {
+							rows.AddRow(msr.id)
+						}
+						mock.ExpectQuery(msr.query).
+							WithArgs(param.Name, param.ConfigFile, param.Value).
+							WillReturnRows(rows)
+					}
+				}
+			}
+
+			mock.ExpectExec("profile_parameter").WillReturnResult(sqlmock.NewResult(1, int64(len(testCase.parameters))))
+
+			newParams, existingParams, err := importProfileParameters(int64(1), testCase.parameters, db.MustBegin().Tx)
```

Comment:
`Cannot use int64(1) (type int64) as type int in argument to importProfileParameters`